src/test/obj_tx_lock/obj_tx_lock.c

```diff
@@ -165,6 +166,22 @@ do_tx_add_taken_lock(struct transaction_data *data)
 	return NULL;
 }
 
+static void
+do_fault_injection(struct transaction_data *data)
+{
+	if (!pmemobj_fault_injection_enabled())
+		return;
+
+	pmemobj_inject_fault_at(PMEM_MALLOC, 1, "add_to_tx_and_lock");
+
+	int ret;
+	IS_UNLOCKED(Pop, data->mutexes, data->rwlocks);
+	TX_BEGIN(Pop) {
+		DO_LOCK(data->mutexes, data->rwlocks);
+	} TX_END
+	UT_ASSERTeq(errno, ENOMEM);
```

Comment:
this should be in onabort section with ut_assert(0) in oncommit section, because errno is not set when transaction didn't fail